Kudu.Services.Web/App_Start/NinjectServices.cs

```diff
@@ -720,11 +724,24 @@ private static IEnvironment GetEnvironment(IDeploymentSettingsManager settings =
             string root = PathResolver.ResolveRootPath();
             string siteRoot = Path.Combine(root, Constants.SiteFolder);
             string repositoryPath = Path.Combine(siteRoot, settings == null ? Constants.RepositoryPath : settings.GetRepositoryPath());
+            string binPath = HttpRuntime.BinDirectory;
+
+            if (!string.IsNullOrWhiteSpace(binPath) && !OSDetecter.IsOnWindows())
+            {
+                int binIdx = binPath.LastIndexOf("Bin", StringComparison.Ordinal);
```

Comment:
Concern was that code tests if it finds a match, implying it may not. But if it doesn't end with bin, it may end up finding some unrelated 'bin' in the middle of the path. Shouldn't it test for EndsWith instead?